components/toggle/Toggle.js

```diff
@@ -1,28 +1,61 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { t } from 'ttag';
+import { c } from 'ttag';
 
 import Input from '../input/Input';
-import Label from '../label/Label';
+import Icon from '../icon/Icon';
 
-const Toggle = ({ id }) => {
+const label = (type, key) => {
+    const I18N = {
+        on: c('Toggle button').t`On`,
+        off: c('Toggle button').t`Off`
+    };
+
+    if (type === 'icon') {
+        return (
+            <span className="pm-toggle-label-text">
+                <Icon name={key} alt={I18N[key]} className="pm-toggle-label-img" />
+            </span>
+        );
+    }
+
+    if (key === 'on') {
+        return <span className="pm-toggle-label-text">{I18N[key]}</span>;
+    }
+
+    return <span className="pm-toggle-label-text">{I18N[key]}</span>;
+};
+
+const Toggle = ({ id, checked, onChange, type }) => {
+    const [value, setValue] = useState(checked);
     return (
         <>
-            <Input type="checkbox" id={id} className="pm-toggle-checkbox" />
-            <Label htmlFor={id} className="pm-toggle-label">
-                <span className="pm-toggle-label-text">{t`On`}</span>
-                <span className="pm-toggle-label-text">{t`Off`}</span>
-            </Label>
+            <Input
```

Comment:
You can use the existing `<Checkbox>` component here.

Reply:
Before we can use it we need to fix the CSS as the selector won't work.
Ping @nico3333fr the checkbox is made using `+` our Checkbox component is input + span so it won't work.

Reply:
Quick detail: there is no more text in the toggle component.